pkg/controller/garbagecollector/garbagecollector_test.go

```diff
@@ -213,12 +223,27 @@ func setupGC(t *testing.T, config *restclient.Config) garbageCollector {
 	if err != nil {
 		t.Fatal(err)
 	}
+	tweakableRM := meta.NewDefaultRESTMapper(nil)
+	rm := &testRESTMapper{meta.MultiRESTMapper{tweakableRM, testrestmapper.TestOnlyStaticRESTMapper(legacyscheme.Scheme)}}
 
 	client := fake.NewSimpleClientset()
 	sharedInformers := informers.NewSharedInformerFactory(client, 0)
+	metadataInformers := metadatainformer.NewSharedInformerFactory(metadataClient, 0)
 	alwaysStarted := make(chan struct{})
 	close(alwaysStarted)
-	gc, err := NewGarbageCollector(ctx, client, metadataClient, &testRESTMapper{testrestmapper.TestOnlyStaticRESTMapper(legacyscheme.Scheme)}, ignoredResources, sharedInformers, alwaysStarted)
+
+	dependencyGraphBuilder := NewDependencyGraphBuilder(
+		ctx,
+		metadataClient,
+		alwaysStarted,
+		rm,
+		informerfactory.NewInformerFactory(sharedInformers, metadataInformers),
```

Comment:
this used to be `sharedInformers`?

Reply:
I tried with just sharedInformers, but it wasn't checking resource sync correctly. So I used this approach for the testing. In production code this is coming for controllerContext which works correctly

Reply:
> I tried with just sharedInformers, but it wasn't checking resource sync correctly. So I used this approach for the testing. In production code this is coming for controllerContext which works correctly

This sounds like a problem.  If changes in construction broke capability, then we have trouble.

Reply:
have you debugging this one?  A failure here suggests that our constructors are somehow not equivalent.

Reply:
yup. debugging it now

Reply:
reverted to original. no changes needed here.